compose/ui/ui-text/src/skikoMain/kotlin/androidx/compose/ui/text/platform/SkiaParagraphIntrinsics.skiko.kt

```diff
@@ -82,18 +82,31 @@ internal class SkiaParagraphIntrinsics(
         maxIntrinsicWidth = ceil(para.maxIntrinsicWidth)
     }
 
-    private fun resolveTextDirection(direction: TextDirection?): ResolvedTextDirection {
-        return when (direction) {
+    private fun resolveTextDirection(
+        textDirection: TextDirection? = null,
+        localeList: LocaleList? = null
+    ): ResolvedTextDirection {
+        return when (textDirection ?: TextDirection.Content) {
             TextDirection.Ltr -> ResolvedTextDirection.Ltr
             TextDirection.Rtl -> ResolvedTextDirection.Rtl
-            TextDirection.Content -> contentBasedTextDirection() ?: ResolvedTextDirection.Ltr
-            TextDirection.ContentOrLtr -> contentBasedTextDirection() ?: ResolvedTextDirection.Ltr
-            TextDirection.ContentOrRtl -> contentBasedTextDirection() ?: ResolvedTextDirection.Rtl
-            else -> ResolvedTextDirection.Ltr
+            TextDirection.Content -> contentBasedTextDirection(text) { localeBasedTextDirection(localeList?.firstOrNull()) }
+            TextDirection.ContentOrLtr -> contentBasedTextDirection(text) { ResolvedTextDirection.Ltr }
+            TextDirection.ContentOrRtl -> contentBasedTextDirection(text) { ResolvedTextDirection.Rtl }
+            else -> error("Invalid TextDirection.")
```

Comment:
Previously it was ResolvedTextDirection.Ltr by default.
Maybe better to stay it?

Reply:
No. Current implementation matches Android one. Current fallback is `Content` (line 89), not error.